runtime/runtime-params-estimator/estimator-warehouse/src/estimate.rs

```diff
@@ -0,0 +1,101 @@
+use clap::Parser;
+use nix::unistd::Uid;
+use xshell::{cmd, Shell};
+
+use crate::{db::Db, import::ImportConfig};
+
+/// Additional information required for estimation.
+#[derive(Debug, Parser)]
+pub(crate) struct EstimateConfig {
+    /// Specify the directory of a different repository, if not estimating the current one.
+    #[clap(long)]
+    pub external_repo: Option<String>,
+    /// Specify the directory for near state used by the estimator. Will use
+    /// temporary directory if unspecified.
+    #[clap(long)]
+    pub home: Option<String>,
+}
+
+pub(crate) fn run_estimation(db: &Db, config: &EstimateConfig) -> anyhow::Result<()> {
+    let sh = Shell::new()?;
+
+    let mut maybe_tmp = None;
+
+    let estimator_home = match &config.home {
+        Some(home) => &home,
+        None => {
+            maybe_tmp = Some(tempfile::tempdir()?);
+            maybe_tmp.as_ref().unwrap().path().to_str().unwrap()
+        }
+    };
+
+    if let Some(external_repo) = &config.external_repo {
+        sh.change_dir(external_repo);
+    }
+    let output = cmd!(sh, "git rev-parse --show-toplevel").output()?;
+    let mut git_root = String::from_utf8_lossy(&output.stdout).to_string();
+    git_root.pop(); // \n
```

Comment:
```suggestion
    let output = cmd!(sh, "git rev-parse --show-toplevel").read()?;
```

It does `.pop` thing